### PR TITLE
WP-152: Update basic teaser design in website-builder

### PR DESCRIPTION
**Pull request by antkiewiczk**

### CODE REVIEW

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 1]
```diff
@@ -185,7 +217,68 @@ export const TeaserAuthors = styled('div')`
...
+    font-size: 18px;
+    transform: ${({theme}) => `translateY(-${theme.spacing(3.5)})`};
+  }
+`
+
+export const TeaserTitlesStyled = styled(TeaserTitle)`
+  margin: 0;
+`
+
+export const AuthorsAndDate = styled('p')`
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:269]: Not a paragraph so probably shouldn't be one

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 2]
```diff
@@ -185,7 +217,68 @@ export const TeaserAuthors = styled('div')`
...
+  margin: 0;
+  font-size: 12px;
+`
+
+export const TeaserLeadStyled = styled('div')`
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:274]: a paragraph so probably should be one. You should be able to use the Paragraph from website builder.

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 3]
```diff
@@ -185,7 +217,68 @@ export const TeaserAuthors = styled('div')`
...
+    isHovered ? theme.palette.primary.main : theme.palette.common.black};
+  height: 3px;
+  width: 100%;
+  margin-bottom: ${({theme}) => theme.spacing(1.5)};
+`
+
+export const TeaserPreTitleStyled = styled(TeaserPreTitle)<{isHovered: boolean}>`
+  transition: background-color 0.3s ease-in-out;
+  background-color: ${({theme, isHovered}) =>
+    isHovered ? theme.palette.primary.main : theme.palette.secondary.main};
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:243]: dont think thats part of the design with the hover.

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 4]
```diff
@@ -196,32 +289,52 @@ export const Teaser = ({teaser, alignment, className}: BuilderTeaserProps) => {
...
+  const enhancedImageStyles = useMemo(
+    () => getEnhancedImageStyles(theme, isHovered),
+    [theme, isHovered]
+  )
 
-          <TeaserContent>
+  return (
+    <TeaserWrapper
+      {...alignment}
+      onMouseEnter={() => setIsHovered(true)}
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:304]: This has many accessibility issues and can bug out in multiple ways. You can use react-aria for this or css.

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 5]
```diff
@@ -185,7 +217,68 @@ export const TeaserAuthors = styled('div')`
...
+  width: 100%;
+  margin-bottom: ${({theme}) => theme.spacing(1.5)};
+`
+
+export const PreTitleStyled = styled('span')<{isHovered: boolean}>`
+  transition: background-color 0.3s ease-in-out;
+  padding: ${({theme}) => `${theme.spacing(0.5)} ${theme.spacing(2)}`};
+  background-color: ${({theme, isHovered}) =>
+    isHovered ? theme.palette.primary.main : theme.palette.secondary.main};
+  display: inline-block;
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:254]: spans are inline, make it a div.

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 6]
```diff
@@ -196,32 +289,52 @@ export const Teaser = ({teaser, alignment, className}: BuilderTeaserProps) => {
...
+      <Link color="inherit" href={href ?? ''} className={className} css={linkStyles}>
+        {image ? (
+          <ImageWrapperStyled>
+            <Image image={image} css={enhancedImageStyles} />
+          </ImageWrapperStyled>
+        ) : (
+          <ImagePlaceholder />
+        )}
+
+        <TeaserContentStyled>
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:315]: As I mentioned in the meeting already before, this now causes that you can not reorder the elements at will. Do not have a wrapper around the content. 

If you don't have a wrapper around the content you can use grid-template-area & grid-area to define how things are displayed.

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 7]
```diff
@@ -196,32 +289,52 @@ export const Teaser = ({teaser, alignment, className}: BuilderTeaserProps) => {
...
+
+          <TeaserLeadStyled>{lead}</TeaserLeadStyled>
+
+          <AuthorsAndDate>
+            {authors && authors?.length ? <Authors>Von {authors?.join(', ')} </Authors> : null}
             {publishDate && (
-              <TeaserDate dateTime={publishDate}>
-                {date.format(new Date(publishDate), false)}
-              </TeaserDate>
+              <time dateTime={publishDate} css={{fontWeight: 400}}>
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:331]: no inline css please

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 8]
```diff
@@ -185,7 +217,68 @@ export const TeaserAuthors = styled('div')`
...
+export const TeaserPreTitleNoContent = styled(TeaserPreTitle)<{isHovered: boolean}>`
+  transition: background-color 0.3s ease-in-out;
+  background-color: ${({theme, isHovered}) =>
+    isHovered ? theme.palette.primary.main : theme.palette.common.black};
+  height: 3px;
+  width: 100%;
+  margin-bottom: ${({theme}) => theme.spacing(1.5)};
+`
+
+export const TeaserPreTitleStyled = styled(TeaserPreTitle)<{isHovered: boolean}>`
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:240]: Probably shouldn't have `Styled` postfix

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 9]
```diff
@@ -185,7 +217,68 @@ export const TeaserAuthors = styled('div')`
...
+  aspect-ratio: 1/1;
+`
+
+export const TeaserContentStyled = styled(TeaserContent)`
+  grid-column: 1/13;
+`
+
+export const TeaserPreTitleNoContent = styled(TeaserPreTitle)<{isHovered: boolean}>`
+  transition: background-color 0.3s ease-in-out;
+  background-color: ${({theme, isHovered}) =>
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:233]: dont think hover color style is part of the design

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 10]
```diff
@@ -162,20 +138,76 @@ export const selectTeaserAuthors = (teaser: TeaserType) => {
...
+
+const teaserImageStyles = css`
   width: 100%;
   object-fit: cover;
+  grid-column: 1/13;
+  transition: transform 0.3s ease-in-out;
 `
 
-const teaserLinkStyles = css`
+const getEnhancedImageStyles = (theme: Theme, isHovered: boolean) => css`
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:181]: Probably want to do it like you did before in the navigation

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 11]
```diff
@@ -132,7 +107,8 @@ export const selectTeaserDate = (teaser: TeaserType) => {
       return teaser.page?.publishedAt
     }
 
-    case 'ArticleTeaser': {
+    case 'ArticleTeaser':
+    case 'PeerArticleTeaser': {
```
Comment (Itrulia) [inline libs/block-content/website/src/lib/teaser/teaser.tsx:111]: 👍🏼

File: libs/block-content/website/src/lib/teaser/teaser.tsx (hunk 4 above)
Comment (antkiewiczk) [reply libs/block-content/website/src/lib/teaser/teaser.tsx:304]: I tried using css but due to the structure, using of website-builder components it was impossible to implement.

File: libs/block-content/website/src/lib/teaser/teaser.tsx (hunk 9 above)
Comment (antkiewiczk) [reply libs/block-content/website/src/lib/teaser/teaser.tsx:233]: hover stuff is taken from the note written on the designs, sent you screenshot on slack (uploading on gh doesn't work)

File: libs/block-content/website/src/lib/teaser/teaser.tsx (hunk 10 above)
Comment (Itrulia) [reply libs/block-content/website/src/lib/teaser/teaser.tsx:181]:
```
const useNavbarLinkStyles = () => {
  const theme = useTheme()

  return useMemo(
    () => css`
      font-size: 1rem;
      margin: 0 ${theme.spacing(1)} 0 ${theme.spacing(2)};
      text-decoration: none;
      color: ${theme.palette.common.black};

      ${theme.breakpoints.up('md')} {
        font-size: 1.3rem;
      }
    `,
    [theme]
  )
}
```

Like you done in the navbar. This way it doesn't rerender unnecessarily which is better than my example implementation (basically like this one).

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 12]
```diff
@@ -162,20 +138,76 @@ export const selectTeaserAuthors = (teaser: TeaserType) => {
...
+  grid-template-rows: min-content;
+  align-items: start;
+  text-decoration: none;
 `
 
 export const TeaserTitles = styled('header')``
-export const TeaserTitle = styled('h1')``
-export const TeaserPreTitle = styled('div')``
+export const TeaserTitle = styled('h1')`
+  font-size: 26px;
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply libs/block-content/website/src/lib/teaser/teaser.tsx:200]: Will be done with Tsri

File: libs/block-content/website/src/lib/teaser/teaser.tsx (hunk 4 above)
Comment (Itrulia) [reply libs/block-content/website/src/lib/teaser/teaser.tsx:304]: How so?

Comment (antkiewiczk) [reply libs/block-content/website/src/lib/teaser/teaser.tsx]: ok I used react-aria

File: libs/block-content/website/src/lib/teaser/teaser.tsx [hunk 13]
```diff
@@ -162,20 +138,76 @@ export const selectTeaserAuthors = (teaser: TeaserType) => {
...
+`
+
+export const TeaserInnerWrapper = styled('div')`
+  display: grid;
+  gap: ${({theme}) => theme.spacing(1)};
+  grid-template-rows: auto;
+  grid-auto-rows: max-content;
+`
+
+export const ImageWrapperStyled = styled('div')`
```
[the remark this answers is not in the document]
Comment (antkiewiczk) [reply libs/block-content/website/src/lib/teaser/teaser.tsx:168]: I tried changing it to ImageWrapper, but then it has conflicts with some other ImageWrapper exported from some other file so renamed it to TeaserImageWrapper

**Pull request merged**